site/app/controllers/PollController.php

```diff
@@ -503,6 +506,86 @@ public function openPoll(): RedirectResponse {
     }
 
     /**
+     * @Route("/courses/{_semester}/{_course}/polls/addCustomResponse", methods={"POST"})
+     */
+    public function addCustomResponse(): JsonResponse {
+        $poll_id = intval($_POST['poll_id'] ?? -1);
+        $poll_response = $_POST['custom_response'];
+        $user_id = $this->core->getUser()->getId();
+        $em = $this->core->getCourseEntityManager();
+        $repo = $em->getRepository(Poll::class);
+        /** @var Poll|null */
+        $poll = $repo->findByIDWithOptions($poll_id);
```

Comment:
Why do you need to preload the options here?  Shouldn't you only need to load the poll itself since you're only every adding things to it?